.github/workflows/pull-request.yaml

```diff
@@ -6,6 +6,35 @@ on:
       - main
 
 jobs:
+  version-check:
+    name: Version Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Check All Versions Match
+        run: |
+          pkg_version=$(grep -P -o 'version = "\d+\.\d+\.\d+"' pyproject.toml | grep -P -o '\d+\.\d+\.\d+')
+          rock_version=$(grep -P -o 'version: "\d+\.\d+\.\d+"' rockcraft.yaml | grep -P -o '\d+\.\d+\.\d+')
+          rock_pkg_version=$(grep -P -o 'cos-alerter==\d+\.\d+\.\d+' rockcraft.yaml | grep -P -o '\d+\.\d+\.\d+')
```

Comment:
You may be able to avoid repeating the regex with something similar to https://github.com/canonical/observability/blob/05aae2717ece6b3700a68f365e6bedb321340048/.github/workflows/update-rock.yaml#L79

However, perhaps you could extract by path?
- yq
- [dasel](https://github.com/TomWright/dasel)

Reply:
While this looks very cool, I  am hesitant to use a tool in our workflow that has no apt or snap package.